host_applications/linux/apps/raspicam/RaspiVid.c

```diff
@@ -1089,11 +1090,21 @@ static FILE *open_filename(RASPIVID_STATE *pState, char *filename)
    {
       // Create a new filename string
 
-      // If %d or %u, assume a segment numnber, otherwise use the formatter as
-      // input to strnftime
-      if (strstr(filename,"%u") != NULL || strstr(filename,"%d") != NULL)
-      {
-          asprintf(&tempname, filename, pState->segmentNumber);
+	  //If %d/%u or any valid combination e.g. %04d is specified, assume segment number.
+	  bool bSegmentNumber = false;
+	  const char* pPercent = strstr(filename, "%");
+	  if (pPercent)
+	  {
+	  	pPercent++;
+	  	while (*pPercent != '\0' && isdigit(*pPercent))
```

Comment:
My logic says that '\0' isn't a digit, therefore the ```isdigit``` condition would abort the loop anyway at '\0'.
Whether a compiler agrees with me is another question :-o

Reply:
Good point! https://en.cppreference.com/w/cpp/string/byte/isdigit Suggests isdigit would behave the way you describe.